MSLaunches.Frontend/src/app/lunch/lunch.service.ts

```diff
@@ -53,16 +66,52 @@ export class LunchService {
         return this.fillDates(weekly);
     }
 
-      private mergeUserLunchs(userSelection: Array<UserSelection>, menu: Array<Lunch>): Array<Lunch> {
-        menu.forEach(lunch => {
-            if (userSelection.some(a => a.lunchId === lunch.id)) {
-                lunch.isSelected = true;
+    save(lunch: Lunch): Observable<string> {
+        const userId = JSON.parse(sessionStorage.getItem(this.credentialsKey)).userId;
+        if (!lunch.userLunchId || lunch.userLunchId.length === 0) {
+            return this.httpClient.post(this.routes.getGeneric(userId), this.mapToBackEnd(lunch))
+                .pipe(
+                    map((a: any) => a.userLunchId)
+                );
+        } else {
+            return this.httpClient.put(this.routes.getAccessor(userId, lunch.userLunchId), this.mapToBackEnd(lunch))
+                .pipe(map((a: any) => lunch.userLunchId));
+        }
+    }
```

Comment:
I think we should use only `put` since we are doing changes, why you are using both?

Reply:
If it is the first time the user creates a lunch for him, or if he deletes it, he needs a new record